code/datums/mind.dm

```diff
@@ -1616,6 +1672,27 @@
 		var/obj/effect/proc_holder/spell/S = X
 		S.action.Grant(new_character)
 
+/datum/mind/proc/disrupt_spells(var/delay, var/list/exceptions = New())
+	for(var/obj/effect/proc_holder/spell/S in spell_list)
+		var/found = 0
+		for(var/type in spells)

```

Comment:
You probably meant in exceptions here, and you can just use continue instead of found and break.